chain/chain/src/chain.rs

```diff
@@ -519,20 +522,35 @@ impl Chain {
                 _ => return Err(e),
             },
         };
+
         if gc_stop_height > head.height {
             return Err(ErrorKind::GCError(
                 "gc_stop_height cannot be larger than head.height".into(),
             )
             .into());
         }
+        let prev_epoch_id = self.get_block_header(&head.prev_block_hash)?.epoch_id();
+        let epoch_change = prev_epoch_id != &head.epoch_id;
+        let fork_tail = if epoch_change {
+            let mut chain_store_update = self.store.store_update();
+            chain_store_update.update_fork_tail(gc_stop_height);
```

Comment:
Is it possible that the `fork_tail` hasn't caught up with the previous epoch yet when it got updated to `gc_stop_height`?

Reply:
Yes. It will break nothing if happens once in a period.

We need to make sure that GC works a little bit faster (in average) than the rest of the system creates blocks.